From a546b8779a3c1a631f4af29c53e73da938e15eb9 Mon Sep 17 00:00:00 2001
From: t <tristan75015@gmail.com>
Date: Sun, 11 Oct 2026 05:45:38 +0200
Subject: [PATCH] NonLinearEvolutionProblemImplementation: profiling of the
 post-processings

Sections of the profiler for the execution of the post-processings, the
synchronization of the materials on the host and the execution of each
post-processing.
---
 src/NonLinearEvolutionProblemImplementation.cxx | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/NonLinearEvolutionProblemImplementation.cxx b/src/NonLinearEvolutionProblemImplementation.cxx
index a2bd6650a9..7306c6156c 100644
--- a/src/NonLinearEvolutionProblemImplementation.cxx
+++ b/src/NonLinearEvolutionProblemImplementation.cxx
@@ -923,8 +923,10 @@ namespace mfem_mgis {
 
   bool NonLinearEvolutionProblemImplementation<true>::executePostProcessings(
       Context& ctx, const real t, const real dt) noexcept {
+    CatchTimeSection(ctx, "NLEP::executePostProcessings");
     // the post-processings read the materials on the host
     if (this->mgis_integrator != nullptr) {
+      CatchTimeSection(ctx, "NLEP::PP::synchronizeOnHost");
       try {
         this->mgis_integrator->synchronizeOnHostForReading();
       } catch (...) {
@@ -933,6 +935,7 @@ namespace mfem_mgis {
     }
     auto success = true;
     for (auto& p : this->postprocessings) {
+      CatchTimeSection(ctx, "NLEP::PP::execute");
       if (!p->execute(ctx, *this, t, dt)) {
         success = false;
       }
@@ -1264,8 +1267,10 @@ namespace mfem_mgis {
 
   bool NonLinearEvolutionProblemImplementation<false>::executePostProcessings(
       Context& ctx, const real t, const real dt) noexcept {
+    CatchTimeSection(ctx, "NLEP::executePostProcessings");
     // the post-processings read the materials on the host
     if (this->mgis_integrator != nullptr) {
+      CatchTimeSection(ctx, "NLEP::PP::synchronizeOnHost");
       try {
         this->mgis_integrator->synchronizeOnHostForReading();
       } catch (...) {
@@ -1274,6 +1279,7 @@ namespace mfem_mgis {
     }
     auto success = true;
     for (auto& p : this->postprocessings) {
+      CatchTimeSection(ctx, "NLEP::PP::execute");
       if (!p->execute(ctx, *this, t, dt)) {
         success = false;
       }
-- 
2.43.0

